src/main/java/cz/smarteon/loxone/LoxoneEndpoint.java

```diff
@@ -15,17 +15,34 @@
 public final class LoxoneEndpoint {
 
     private final String address;
-    private final int port;
+    private int port;
     private final boolean useSsl;
+    private boolean isCloud;
+    private String host;
+    private String path;
 
     private static final String WS_TEMPLATE = "%s://%s:%d/ws/rfc6455";
+    private static final String WS_TEMPLATE_DNS = "%s://%s/ws/rfc6455";
```

Comment:
also this pattern is not only related to the DNS loxone cloud... but also other situations - adjust naming